cranelift/codegen/src/isa/x86/unwind/systemv.rs

```diff
@@ -0,0 +1,485 @@
+            frame_register,
+            instructions: Vec::new(),
+            stack_size: None,
+            epilogue_pop_offsets: Vec::new(),
+        }
+    }
+
+    fn push_reg(&mut self, offset: u32, arg: Value) {
+        self.cfa_offset += 8;
+
+        let reg = self.func.locations[arg].unwrap_reg();
+
+        // Update the CFA if this is the save of the frame pointer register or if a frame pointer isn't being used
+        // When using a frame pointer, we only need to update the CFA to account for the push of the frame pointer itself
+        if match self.frame_register {
+            Some(fp) => reg == fp,
+            None => true,
+        } {
+            self.instructions
+                .push((offset, CallFrameInstruction::CfaOffset(self.cfa_offset)));
+        }
+
+        // Pushes in the prologue are register saves, so record an offset of the save
+        self.instructions.push((
+            offset,
+            CallFrameInstruction::Offset(
+                map_reg(self.isa, reg)
+                    .expect("a register mapping from cranelift to gimli")
+                    .0,
+                -self.cfa_offset,
+            ),
+        ));
+    }
+
+    fn adjust_sp_down(&mut self, offset: u32) {
+        // Don't adjust the CFA if we're using a frame pointer
+        if self.frame_register.is_some() {
+            return;
+        }
+
+        self.cfa_offset += self
+            .stack_size
+            .expect("expected a previous stack size instruction");
+        self.instructions
+            .push((offset, CallFrameInstruction::CfaOffset(self.cfa_offset)));
+    }
+
+    fn adjust_sp_down_imm(&mut self, offset: u32, imm: i64) {
+        assert!(imm <= core::u32::MAX as i64);
+
+        // Don't adjust the CFA if we're using a frame pointer
+        if self.frame_register.is_some() {
+            return;
+        }
+
+        self.cfa_offset += imm as i32;
+        self.instructions
+            .push((offset, CallFrameInstruction::CfaOffset(self.cfa_offset)));
+    }
+
+    fn adjust_sp_up_imm(&mut self, offset: u32, imm: i64) {
+        assert!(imm <= core::u32::MAX as i64);
+
+        // Don't adjust the CFA if we're using a frame pointer
+        if self.frame_register.is_some() {
+            return;
+        }
+
+        self.cfa_offset -= imm as i32;
+        self.instructions
+            .push((offset, CallFrameInstruction::CfaOffset(self.cfa_offset)));
+    }
+
+    fn move_reg(&mut self, offset: u32, src: RegUnit, dst: RegUnit) {
+        if let Some(fp) = self.frame_register {
+            // Check for change in CFA register (RSP is always the starting CFA)
+            if src == (RU::rsp as RegUnit) && dst == fp {
+                self.instructions.push((
+                    offset,
+                    CallFrameInstruction::CfaRegister(
+                        map_reg(self.isa, dst)
+                            .expect("a register mapping from cranelift to gimli")
```

Comment:
Here and near `map_reg` in this file, we need to avoid `panic`. It maybe reasonable to just skip the operation if register is unknown to DWARF (I don't think we can create a fallback `CallFrameInstruction`).

Reply:
Good point.  We could also return an error instead of panicking since the creation function now returns `CodegenResult` from #1216.

Reply:
`map_reg` should now return a `Result` since #1451 so it should be a bit easier; prior to that PR `map_reg` panicked so when I added `RegisterMappingError` I kept the same behavior with this `expect` (and one other place I believe). I agree it would be better if we avoided panicking!

Reply:
I missed that will all the merging.  I'll get that fixed.

Reply:
I've pushed up a commit to remove the `expect` from `map_reg` calls.

Reply:
Thanks. FWIW I think generating even incomplete FDE for backtrace outweighs maintainability/diagnostics of unwind generation code. That said, we may ignore map_reg errors in future.